aten/src/ATen/native/cpu/Activation.cpp

```diff
@@ -314,6 +314,65 @@ void GeluBackwardKernelImpl(TensorIterator& it) {
   }
 }
 
+void hardsigmoid_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "hardsigmoid_cpu", [&] {
+    const scalar_t zero(0.0f);
+    const scalar_t three(3.0f);
+    const scalar_t six(6.0f);
+    using Vec = vec256::Vec256<scalar_t>;
+    const Vec kZeroVec(0);
+    const Vec kThreeVec(3);
+    const Vec kSixVec(6);
+    cpu_kernel_vec(
+        iter,
+        [&](scalar_t self_val) {
+          scalar_t val_plus_three = self_val + three;
+          scalar_t val_plus_three_gt_zero = val_plus_three > zero
+            ? val_plus_three
+            : zero;
+          return (
+            val_plus_three_gt_zero < six ? val_plus_three_gt_zero : six
+          ) / six;
```

Comment:
Can't this be implemented branch-free like `std::min(std::max(x + three, zero), size) / six`